suitcase/fields.py

```diff
@@ -615,7 +615,7 @@ def pack(self, stream):
     def unpack(self, data):
         # length of data will be determined by bytes_required output value
         # which is in turn determined by our condition evaluation
-        if len(data) > 0:
+        if self.condition(self._parent):

```

Comment:
The comment lines above this line of code should be removed as it no longer is applicable to the new contents.